router/core/request_compression_test.go

```diff
@@ -0,0 +1,207 @@
+package core
+
+import (
+	"compress/flate"
+	"compress/gzip"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	br "github.com/andybalholm/brotli"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+var responses = map[string]string{
+	"text/html":                "<html><body>Hello World !</body></html>",
+	"text/css":                 "body { color: black; }",
+	"text/plain":               "Hello World !",
+	"text/javascript":          "console.log('Hello World !');",
+	"application/javascript":   "console.log('Hello World !');",
+	"application/x-javascript": "console.log('Hello World !');",
+	"application/json":         `{"message": "Hello World !"}`,
+	"application/atom+xml":     `<feed><title>Hello World !</title></feed>`,
+	"application/rss+xml":      `<rss><channel><title>Hello World !</title></channel></rss>`,
+	"image/svg+xml":            `<svg><text>Hello World !</text></svg>`,
+	"application/graphql":      `query { message }`,
+}
+
+func TestCompression(t *testing.T) {
+	httpRouter := chi.NewRouter()
+
+	// Add a route for each content type
+	for _, contentType := range CustomCompressibleContentTypes {
+		httpRouter.Get("/"+contentType, func(contentType string) http.HandlerFunc {
+			return func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Add("Content-Type", contentType)
+				w.Write([]byte(responses[contentType]))
+			}
+		}(contentType))
+	}
+
+	// Create a test server without compression
+	ts := httptest.NewServer(httpRouter)
+	defer ts.Close()
+
+	// Test each content type without compression
+	for _, contentType := range CustomCompressibleContentTypes {
+		t.Run("uncompressed-"+contentType, func(t *testing.T) {
+			testUncompressedResponse(t, ts.URL+"/"+contentType, contentType)
+		})
+	}
+
+	// Now add compression middleware
+	compressedRouter := chi.NewRouter()
+
+	// Adds Brotli compressor
+	brCompressor := middleware.NewCompressor(5, CustomCompressibleContentTypes...)
+	brCompressor.SetEncoder("br", func(w io.Writer, level int) io.Writer {
+		return br.NewWriterLevel(w, level)
+	})
+	compressedRouter.Use(middleware.AllowContentEncoding("deflate", "gzip"))
```

Comment:
@Rutik7066 here why is "br" not part of the list?

Reply:
deflate and gzip are supported by go-chi router. so we have to add brotli compression as custom middleware.

Reply:
How is it related to the middleware that validates what content-encoding is allowed?

Reply:
Yes your are right . it should be before compressor middleware and br should be added in the list